samples/bluetooth/peripheral_proximity/prj_minimal.conf

```diff
@@ -0,0 +1,115 @@
+# BEGIN Configurations from basic/minimal
+
+CONFIG_NCS_SAMPLES_DEFAULTS=y
+
+CONFIG_BT=y
+CONFIG_BT_PERIPHERAL=y
+CONFIG_BT_DEVICE_NAME="Nordic_Proximity"
+
+# Enable the service
+CONFIG_BT_LLS=y
+CONFIG_BT_IAS=y
+
+CONFIG_DK_LIBRARY=y
+
+# Drivers and peripherals
+CONFIG_I2C=n
+CONFIG_WATCHDOG=n
+CONFIG_GPIO=n
+CONFIG_PINMUX=n
+CONFIG_SPI=n
+CONFIG_SERIAL=n
+
+# Power management
+CONFIG_PM=n
+
+# Interrupts
+CONFIG_DYNAMIC_INTERRUPTS=n
+CONFIG_IRQ_OFFLOAD=n
+
+# Memory protection
+CONFIG_THREAD_STACK_INFO=n
+CONFIG_THREAD_CUSTOM_DATA=n
+CONFIG_FPU=n
+
+# Boot
+CONFIG_BOOT_BANNER=n
+CONFIG_BOOT_DELAY=0
+
+# Console
+CONFIG_CONSOLE=n
+CONFIG_UART_CONSOLE=n
+CONFIG_STDOUT_CONSOLE=n
+CONFIG_PRINTK=n
+CONFIG_EARLY_CONSOLE=n
+
+# Build
+CONFIG_SIZE_OPTIMIZATIONS=y
+
+# ARM
+CONFIG_ARM_MPU=n
+# END Configurations from basic/minimal
+
+# In order to correctly tune the stack sizes for the threads the following
+# Configurations can enabled to print the current use:
+# CONFIG_THREAD_NAME=y
+# CONFIG_THREAD_ANALYZER=y
+# CONFIG_THREAD_ANALYZER_AUTO=y
+# CONFIG_THREAD_ANALYZER_RUN_UNLOCKED=y
+# CONFIG_THREAD_ANALYZER_USE_PRINTK=y
+# CONFIG_CONSOLE=y
+# CONFIG_UART_CONSOLE=y
+# CONFIG_SERIAL=y
+# CONFIG_PRINTK=y
+
+# Example output of thread analyzer
+# SDC RX              : unused 816 usage 208 / 1024 (20 %)
+# BT RX               : unused 1784 usage 416 / 2200 (18 %)
+# BT TX               : unused 992 usage 544 / 1536 (35 %)
+# thread_analyzer     : unused 136 usage 376 / 512 (73 %)
+# sysworkq            : unused 1824 usage 224 / 2048 (10 %)
+# MPSL signal         : unused 520 usage 504 / 1024 (49 %)
+# idle 00             : unused 192 usage 64 / 256 (25 %)
+# main                : unused 136 usage 888 / 1024 (86 %)
+# Configurations set based on thread analyzer output.
+CONFIG_SDC_RX_STACK_SIZE=324
+CONFIG_BT_RX_STACK_SIZE=1024
+CONFIG_BT_HCI_TX_STACK_SIZE_WITH_PROMPT=y
+CONFIG_BT_HCI_TX_STACK_SIZE=640
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=256
+CONFIG_MPSL_SIGNAL_STACK_SIZE=640
+CONFIG_MAIN_STACK_SIZE=1024
+CONFIG_IDLE_STACK_SIZE=128
+CONFIG_ISR_STACK_SIZE=1024
+
+# Disable features not needed
+CONFIG_TIMESLICING=n
+CONFIG_MINIMAL_LIBC_MALLOC=n
+CONFIG_LOG=n
+CONFIG_ASSERT=n
```

Comment:
I know it's also disabled in the `prj_minimal.conf` of the peripheral_hr sample, but I wouldn't disable asserts, as some stack functions depend on it for parameter checking. At the very least, I would leave them in, but commented out, so the user knows it's possible to reduce size at the expense of safety. e.g.:
```
# Disabling asserts might cause undefined behavior,
# only disable if truly necessary
# CONFIG_ASSERT=n
# CONFIG_BT_ASSERT=n
```